src/MongoDB/ServerDescription.c

```diff
@@ -0,0 +1,269 @@
+	PHP_ME(ServerDescription, getHelloResponse, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getHost, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getLastUpdateTime, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getPort, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getRoundTripTime, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerDescription, getType, ai_ServerDescription_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	ZEND_NAMED_ME(__construct, PHP_FN(MongoDB_disabled___construct), ai_ServerDescription_void, ZEND_ACC_PRIVATE | ZEND_ACC_FINAL)
+	PHP_FE_END
+	/* clang-format on */
+};
+/* }}} */
+
+/* {{{ MongoDB\Driver\ServerDescription object handlers */
+static zend_object_handlers php_phongo_handler_serverdescription;
+
+static void php_phongo_serverdescription_free_object(zend_object* object) /* {{{ */
+{
+	php_phongo_serverdescription_t* intern = Z_OBJ_SERVERDESCRIPTION(object);
+
+	zend_object_std_dtor(&intern->std);
+
+	if (intern->properties) {
+		zend_hash_destroy(intern->properties);
+		FREE_HASHTABLE(intern->properties);
+	}
+
+	if (intern->server_description) {
+		mongoc_server_description_destroy(intern->server_description);
+	}
+}
+
+static zend_object* php_phongo_serverdescription_create_object(zend_class_entry* class_type) /* {{{ */
+{
+	php_phongo_serverdescription_t* intern = NULL;
+
+	intern = PHONGO_ALLOC_OBJECT_T(php_phongo_serverdescription_t, class_type);
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
+
+	intern->std.handlers = &php_phongo_handler_serverdescription;
+
+	return &intern->std;
+} /* }}} */
+
+HashTable* php_phongo_serverdescription_get_properties_hash(phongo_compat_object_handler_type* object, bool is_debug) /* {{{ */
+{
+	php_phongo_serverdescription_t* intern = NULL;
+	HashTable*                      props;
+
+	intern = Z_OBJ_SERVERDESCRIPTION(PHONGO_COMPAT_GET_OBJ(object));
+
+	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_debug, intern, props, 2);
+
+	{
+		zval                host, port, type;
+		mongoc_host_list_t* host_list = mongoc_server_description_host(intern->server_description);
+
+		ZVAL_STRING(&host, host_list->host);
+		zend_hash_str_update(props, "host", sizeof("host") - 1, &host);
+
+		ZVAL_LONG(&port, host_list->port);
+		zend_hash_str_update(props, "port", sizeof("port") - 1, &port);
+
+		ZVAL_LONG(&type, php_phongo_server_description_type(intern->server_description));
+		zend_hash_str_update(props, "type", sizeof("type") - 1, &type);
+	}
+
+	{
+		const bson_t*         hello_response = mongoc_server_description_hello_response(intern->server_description);
+		php_phongo_bson_state state;
+
+		PHONGO_BSON_INIT_DEBUG_STATE(state);
+
+		if (!php_phongo_bson_to_zval_ex(bson_get_data(hello_response), hello_response->len, &state)) {
+			return false;
```

Comment:
Since `false` is a zero value in C, like `NULL`, this likely doesn't raise a compiler warning; however, it's inconsistent since we do document the function as returning a pointer.

This may actually leak since you're not cleaning up `&state.zchild)`, as is done in other places where `php_phongo_bson_to_zval_ex` may return false (including this PR) earlier in this PR).

contrast this with `php_phongo_readpreference_get_properties_hash`, which also handles an error case. There, we clean up and immediately `goto done;` to `return props` (in whatever state it happens to be in). I believe it does get disregarded on account of the exception. Offhand, I'm not sure if there's a reason we do that vs. returning `NULL` -- but rather than dig through the git history (or PHP internals) to unearth the reason for doing so I think we can just follow convention here.